### PR TITLE
fix: itests: Fix flaky paych test

### PR DESCRIPTION
**Pull request by magik6k**
## Related Issues
Did we really not have an issue for this?

## Proposed Changes
Fix the test

## Additional Info
The test was failing likely due to it running on two nodes, and in some cases when a message creating a channel landed on one node, we'd query state from the other node - which isn't the block miner, and could fall behind in sync (block time was 5ms, so that could happen quite easily). The fix is to wait for the nodes to be in sync at certain points.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [x] Commits have a clear commit message.
- [x] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [x] CI is green

### CODE REVIEW

**Comment by arajasek** [comment]
> ## Related Issues
> Did we really not have an issue for this?
> 

https://github.com/filecoin-project/lotus/issues/9883?

File: itests/paych_api_test.go [hunk 1]
```diff
@@ -55,6 +55,15 @@ func TestPaymentChannelsAPI(t *testing.T) {
 	bms := ens.BeginMiningMustPost(blockTime)
 	bm := bms[0]
 
+	waitRecvInSync := func() {
+		// paymentCreator is the block miner, in some cases paymentReceiver may fall behind, so we wait for it to catch up
+
+		head, err := paymentReceiver.ChainHead(ctx)
+		require.NoError(t, err)
+
+		paymentReceiver.WaitTillChain(ctx, kit.HeightAtLeast(head.Height()))
```
Comment (arajasek) [inline itests/paych_api_test.go:64]: I'd even suggest doing head + 5 here, just in case of minor reorgs.

**Review by arajasek — APPROVED** [review]
I think this makes sense, though I don't know for certain this was the cause of (some of)  the flakiness. I suspect a similar pattern might cause flakiness in other tests too.

File: itests/paych_api_test.go (hunk 1 above)
Comment (magik6k) [reply itests/paych_api_test.go:64]: There's one miner in that test, can't have reorgs

**Pull request merged**